File: append_arb.dart
import 'dart:io';

void main() async {
  final planFile = File('docs/superpowers/plans/2026-07-23-beauty-fix-f-beauty-ui.md');
  final content = await planFile.readAsString();

  // Find EN block
  const enStartStr = 'Change the line `"@onboardingValidationAgeMax": {}` to end with a comma, then insert the entire block below immediately before the final `}`:\n\n  ```json\n';
  final enStartIdx = content.indexOf(enStartStr);
  if (enStartIdx == -1) {
    print('Could not find EN block');
    return;
  }
  final enBlockStart = enStartIdx + enStartStr.length;
  final enBlockEnd = content.indexOf('  ```', enBlockStart);
  final enBlock = content.substring(enBlockStart, enBlockEnd);

  final enArbFile = File('lib/l10n/app_en.arb');
  final enArb = await enArbFile.readAsString();
  final enArbNew = enArb.replaceAll('"@onboardingValidationAgeMax": {}', enBlock);
  await enArbFile.writeAsString(enArbNew);
  print('Updated app_en.arb');

  // Find FR block
  const frStartStr = 'Same mechanic: change the final `"@onboardingValidationAgeMax": {}` to end with a comma and insert this block (same keys, same order, French values — for `beautyCheckin*` this is a **translation** of the English text added in Step 1, per the finding, since this file was originally hardcoded in English while every sibling Beauty screen was hardcoded in French):\n\n  ```json\n';
  var frStartIdx = content.indexOf(frStartStr);
  if (frStartIdx == -1) {
    print('Could not find FR block exactly, using fallback search.');
    // Let's use RegExp for FR block
    final frRegex = RegExp(r'change the final `"@onboardingValidationAgeMax": \{\}` to end with a comma and insert this block.*?:\s*```json\n(.*?)\n\s*```', dotAll: true);
    final frMatch = frRegex.firstMatch(content);
    if (frMatch != null) {
      final frBlock = frMatch.group(1)!;
      final frArbFile = File('lib/l10n/app_fr.arb');
      final frArb = await frArbFile.readAsString();
      final frArbNew = frArb.replaceAll('"@onboardingValidationAgeMax": {}', frBlock);
      await frArbFile.writeAsString(frArbNew);
      print('Updated app_fr.arb via regex fallback');
    } else {
      print('Could not find FR block with fallback');
    }
  } else {
    final frBlockStart = frStartIdx + frStartStr.length;
    final frBlockEnd = content.indexOf('  ```', frBlockStart);
    final frBlock = content.substring(frBlockStart, frBlockEnd);

    final frArbFile = File('lib/l10n/app_fr.arb');
    final frArb = await frArbFile.readAsString();
    final frArbNew = frArb.replaceAll('"@onboardingValidationAgeMax": {}', frBlock);
    await frArbFile.writeAsString(frArbNew);
    print('Updated app_fr.arb');
  }
}

File: append_arb.py
import os
import re
import json

def extract_and_append():
    with open('docs/superpowers/plans/2026-07-23-beauty-fix-f-beauty-ui.md', 'r', encoding='utf-8') as f:
        content = f.read()

    # Find Step 1 block for app_en.arb
    en_pattern = r'Change the line `"@onboardingValidationAgeMax": \{\}` to end with a comma, then insert the entire block below immediately before the final `\}`:\s*```json\s*(.*?)\s*```'
    en_match = re.search(en_pattern, content, re.DOTALL)
    if not en_match:
        print("Could not find EN block")
        return

    en_block = en_match.group(1)
    # The block contains the first line `"onboardingValidationAgeMax": "Age must be at most 100",` which we might not want to duplicate if we just replace
    
    with open('lib/l10n/app_en.arb', 'r', encoding='utf-8') as f:
        en_arb = f.read()
    
    en_arb_new = en_arb.replace('"@onboardingValidationAgeMax": {}', en_block)
    with open('lib/l10n/app_en.arb', 'w', encoding='utf-8') as f:
        f.write(en_arb_new)
        print("Updated app_en.arb")

    # Find Step 2 block for app_fr.arb
    fr_pattern = r'Same mechanic: change the final `"@onboardingValidationAgeMax": \{\}` to end with a comma and insert this block.*?:\s*```json\s*(.*?)\s*```'
    fr_match = re.search(fr_pattern, content, re.DOTALL)
    if not fr_match:
        print("Could not find FR block")
        return

    fr_block = fr_match.group(1)
    with open('lib/l10n/app_fr.arb', 'r', encoding='utf-8') as f:
        fr_arb = f.read()

    fr_arb_new = fr_arb.replace('"@onboardingValidationAgeMax": {}', fr_block)
    with open('lib/l10n/app_fr.arb', 'w', encoding='utf-8') as f:
        f.write(fr_arb_new)
        print("Updated app_fr.arb")

extract_and_append()
